Sources/Auth/Core/AuthProvider.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AuthProvider.swift
+//  PovioKit
+//
+//  Created by Borut Tomazin on 25/11/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+public protocol AuthProvidable {
```

Comment:
> A suggestion for a different naming convention. How about instead of 'AuthProvider' we name it 'Authenticator'? Hence, 'AppleAuthenticator' 'FacebookAuthenticator', ...

I kind a like it. I have other in mind, *.Authorizer. Let's do the voting....
> 
> 
> Wouldn't the API be much nicer if the methods are async / return a Promise, instead of delegate pattern?

Done
> 
> 
> I'm not conviced if this protocol is useful or not. The issue is that each specific auth provider can have different parameters for its signing process. This is obvious by looking into Apple provider, which accepts a 'nonce' parameter and Facebook provider, which requires login permissions. Because this protocol is very rigid, the permissions in Facebook provider have to be hardcoded in the class itself (or modifiable through config). Future providers might have more specific parameters still.
> 
> I propose we get rid of this protocol and allow each of the providers to provide its own, clean, API.

I had a general solution in mind to have a central "manager" that will handle all kind of auths. But as soon as you get to a point where you need to isolate packages due to heavy dependencies, it all falls apart.